pkg/gameserversets/controller.go

```diff
@@ -66,6 +66,10 @@ const (
 
 	// maxPodPendingCount is the maximum number of pending pods per game server set
 	maxPodPendingCount = 5000
+
+	// gameServerErrorDeletionDelay is the minimum amount of time to delay the deletion
+	// of a GameServer in Error state.
+	gameServerErrorDeletionDelay = 10 * time.Second
```

Comment:
Is 10 enough? Should it be bigger? 30?

Reply:
10s is the minimum time, in testing they tended to live a little longer, depending on how busy the controller was (12s-22s). Happy to increase it if you see value there, but I do think there is a balance to be struck here.

Reply:
My only thought was for invalid Podspec GameServers (which is harder to do these days), 30+ seconds is a bit easier to get a `kubectl describe gameserver <name>` than 10-20.

Is there a downside to a longer time?

Reply:
My thought is that as Error GameServers are considered up and pending, there is some portion of the capacity that is blocked. In the case of hitting a ResourceQuota and then going below the ResourceQuota again, it will take longer for that capacity to become available. That for me is the balance. I guess 30s would not be destructive here though, so happy with that change.

Reply:
```suggestion
	gameServerErrorDeletionDelay = 30 * time.Second
```